core/iwasm/common/wasm_runtime_common.c

```diff
@@ -3746,7 +4079,8 @@ wasm_externref_ref2obj(uint32 externref_idx, void **p_extern_obj)
     ExternRefMapNode *node;
 
     if (externref_idx == NULL_REF) {
-        return false;
+        *p_extern_obj = NULL;
```

Comment:
If the object is NULL, will it be converted into NULL_REF in obj2ref?

Reply:
The code is trying to catch a variable initialized with `ref.null`. 

if the object is NULL, we shall give it a reference id instead of NULL_REF.